lib/pages/characters.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../data/bloc/rick_and_morty_bloc.dart';
+import '../data/models/character.dart';
+import '../widgets/character_item.dart';
+import '../widgets/preloading_indicator.dart';
+
+class CharactersPage extends StatefulWidget {
+  @override
+  _CharactersPageState createState() => _CharactersPageState();
+}
+
+class _CharactersPageState extends State<CharactersPage> {
+  final List<Character> _list = [];
+
+  bool newPageError = false;
```

Comment:
this variable looks a bit ugly: you use it to pass some data from your `listen` to your `build`, while in fact they both should just rely on the state. This happen because your state currently represents what is happening "right now" and on the UI you need some information about what happened "before" (error occured while fetching data, for example). This can be fixed easily by adjusting your state, no need to use extra variables